### DISCUSSION TITLE
export raster with replaced values

### DISCUSSION THREAD

**Discussion by wxinl** [corteva/rioxarray#399] [Q&A]
I used the rioxarray to read a dem tiff, then I tried to replace the raster values with a numpy array of the same shape and output the raster with replaced values using rio.to_raster. But when I read the new file, the values are all 0s. Image attached.

![image](https://user-images.githubusercontent.com/48371182/133561568-1c935ae3-c9b4-41c0-b721-99c53df02cb2.png)

**Comment by snowman2** [comment]
@wxinl if you wouldn't mind making these changes, it would better enable us to help you:
- the input raster
- the code as text instead of an image for re-reproducibility

Also, what were the values contained by `dem_rx` right before writing to a raster?

UPDATE:
```python
dem_rx.encoding
```
```
{'dtype': 'int16',
 '_FillValue': -32768.0,
 'grid_mapping': 'spatial_ref',
 'source': './ayt08_nasadem.tif',
 'rasterio_dtype': 'int16'}
```
Since you have a different dtype, you need to do:
```python
dem_rx.encoding = {}
```
Then it should work as expected,

**Reply by wxinl** [reply]
[slope.zip](https://github.com/corteva/rioxarray/files/7180210/slope.zip)
Hi, thank you. The dem_rx contains the elevation of each pixel. A sample file and code are provided.

**Reply by snowman2** [reply]
Can you also provide what the values contained by dem_rx were right before writing to a raster?

**Reply by wxinl** [reply]
It's the slope calculated in the code.
![image](https://user-images.githubusercontent.com/48371182/133674115-f8a5e7fc-507f-4a95-8027-0f1737f61756.png)

**Reply by snowman2** [reply]
Updated the answer.
